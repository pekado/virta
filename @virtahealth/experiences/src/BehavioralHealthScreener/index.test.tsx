import * as React from "react";
import { render, waitFor } from "@testing-library/react-native";
import { testClient, WrapComponentWithVirtaContext } from "../test";
import { BehavioralHealthScreener } from "./index";

describe("Behavioral Health Screener", () => {
  it("renders", async () => {
    testClient.get.mockReturnValueOnce(
      Promise.resolve(mockBehavioralHealthQuestionnaire)
    );
    const screener = render(
      <WrapComponentWithVirtaContext>
        <BehavioralHealthScreener
          virtaId={"123abc"}
          patientActionId={"mock-patient-action-id"}
          fhirQuestionnaireId={"mock-fhir-questionnaire-id"}
          imageMap={{}}
        />
      </WrapComponentWithVirtaContext>
    );

    await waitFor(() => expect(testClient.get).toBeCalled());

    screener.update(
      <WrapComponentWithVirtaContext>
        <BehavioralHealthScreener
          virtaId={"123abc"}
          patientActionId={"mock-patient-action-id"}
          fhirQuestionnaireId={"mock-fhir-questionnaire-id"}
          imageMap={{}}
        />
      </WrapComponentWithVirtaContext>
    );

    // TODO - this should work but doesn't due to a bug
    // https://github.com/callstack/react-native-testing-library/issues/553
    // once the bug is fixed, uncomment this test
    // await waitFor(() => {
    //   screener.queryByText("Self-Reflection");
    // });
    expect(screener).toMatchSnapshot();
  });
});

// eslint-disable-next-line jest/no-export
export const mockBehavioralHealthQuestionnaire = {
  contained: [
    {
      valueSet: {
        compose: {
          include: [
            {
              concept: [
                {
                  code: {
                    value: "confidence_0",
                  },
                  display: {
                    value: "I am not at all confident",
                  },
                },
                {
                  code: {
                    value: "confidence_1",
                  },
                  display: {
                    value: "I am a little confident",
                  },
                },
                {
                  code: {
                    value: "confidence_2",
                  },
                  display: {
                    value: "I am somewhat confident",
                  },
                },
                {
                  code: {
                    value: "confidence_3",
                  },
                  display: {
                    value: "I am very confident",
                  },
                },
              ],
            },
          ],
        },
        description: {
          value: "Confidence Scale",
        },
        extensible: {},
        id: {
          value: "LikertConfidence",
        },
        immutable: {
          value: true,
        },
        name: {
          value: "Confidence Scale",
        },
        status: {
          value: "ACTIVE",
        },
      },
    },
    {
      valueSet: {
        compose: {
          include: [
            {
              concept: [
                {
                  code: {
                    value: "frequency_0",
                  },
                  display: {
                    value: "Always",
                  },
                },
                {
                  code: {
                    value: "frequency_1",
                  },
                  display: {
                    value: "Often",
                  },
                },
                {
                  code: {
                    value: "frequency_2",
                  },
                  display: {
                    value: "Sometimes",
                  },
                },
                {
                  code: {
                    value: "frequency_3",
                  },
                  display: {
                    value: "Never",
                  },
                },
              ],
            },
          ],
        },
        description: {
          value: "Frequency Scale",
        },
        extensible: {},
        id: {
          value: "LikertFrequency",
        },
        immutable: {
          value: true,
        },
        name: {
          value: "Frequency Scale",
        },
        status: {
          value: "ACTIVE",
        },
      },
    },
  ],
  id: {
    value: "21ba9359-44bb-40d7-8dbe-3c49550e2af5",
  },
  item: [
    {
      code: [
        {
          code: {
            value: "self-reflection/self-efficacy-general",
          },
          display: {
            value: "Self-efficacy (General)",
          },
          system: {
            value: "Virta Health",
          },
        },
      ],
      item: [
        {
          code: [
            {
              code: {
                value: "abc",
              },
              display: {
                value: "i can manage",
              },
              system: {
                value: "http://loinc.org",
              },
            },
          ],
          linkId: {
            value: "0.1",
          },
          options: {
            fragment: {
              value: "LikertConfidence",
            },
          },
          required: {
            value: true,
          },
          text: {
            value:
              "I can manage to solve difficult problems if I try hard enough.",
          },
          type: {
            value: "CHOICE",
          },
        },
      ],
      linkId: {
        value: "0",
      },
      required: {
        value: true,
      },
      text: {
        value:
          "Rate your overall level of confidence based on the following sentence:",
      },
      type: {
        value: "GROUP",
      },
    },
    {
      code: [
        {
          code: {
            value: "self-reflection/self-efficacy-managing-emotions",
          },
          display: {
            value: "Self-efficacy (Managing Emotions)",
          },
          system: {
            value: "Virta Health",
          },
        },
      ],
      item: [
        {
          code: [
            {
              code: {
                value: "def",
              },
              display: {
                value: "unexpected events",
              },
              system: {
                value: "http://loinc.org",
              },
            },
          ],
          linkId: {
            value: "1.1",
          },
          options: {
            fragment: {
              value: "LikertConfidence",
            },
          },
          required: {
            value: true,
          },
          text: {
            value:
              "I am confident that I could deal efficiently with unexpected events.",
          },
          type: {
            value: "CHOICE",
          },
        },
      ],
      linkId: {
        value: "1",
      },
      required: {
        value: true,
      },
      text: {
        value:
          "Rate your overall level of confidence based on the following sentence:",
      },
      type: {
        value: "GROUP",
      },
    },
    {
      code: [
        {
          code: {
            value: "self-reflection/perceived-control-over-eating",
          },
          display: {
            value: "Perceived control over eating",
          },
          system: {
            value: "Virta Health",
          },
        },
      ],
      item: [
        {
          code: [
            {
              code: {
                value: "ghi",
              },
              display: {
                value: "cravings",
              },
              system: {
                value: "http://loinc.org",
              },
            },
          ],
          linkId: {
            value: "2.1",
          },
          options: {
            fragment: {
              value: "LikertFrequency",
            },
          },
          required: {
            value: true,
          },
          text: {
            value:
              "How often did you go out of your way to get the food you were craving?",
          },
          type: {
            value: "CHOICE",
          },
        },
      ],
      linkId: {
        value: "2",
      },
      required: {
        value: true,
      },
      text: {
        value: "On average during the past 4 weeksâ€¦.",
      },
      type: {
        value: "GROUP",
      },
    },
  ],
  meta: {
    lastUpdated: {
      precision: "MICROSECOND",
      timezone: "+00:00",
      valueUs: "1598317684291142",
    },
    profile: [
      {
        value:
          "http://hl7.org/fhir/us/sdc/StructureDefinition/sdc-questionnaire|2.0",
      },
    ],
    versionId: {
      value: "MTU5ODMxNzY4NDI5MTE0MjAwMA",
    },
  },
  status: {
    value: "DRAFT",
  },
  title: {
    value: "Self-Reflection",
  },
  publisher: {
    value: "Virta Health",
  },
};

// eslint-disable-next-line jest/no-export
export const mockBehavioralHealthFollowUp = {
  actionTitle: "Focus on emotional agility",
  activityTitle: "Emotional agility",
  activityUrl: "https://en.wikipedia.org/",
};
