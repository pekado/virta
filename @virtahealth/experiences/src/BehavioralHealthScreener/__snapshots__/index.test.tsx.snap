// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Behavioral Health Screener renders 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "marginTop": 25,
      },
    ]
  }
>
  <View
    height={1334}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderBottomColor": "#e6e8eb",
          "borderBottomWidth": 1,
          "display": "flex",
          "flexShrink": 0,
          "height": 80,
          "justifyContent": "center",
          "width": "100%",
          "zIndex": 2,
        },
      ]
    }
    width={750}
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgb(36, 38, 43)",
            "fontFamily": "Whitney-Medium",
            "fontSize": 20,
            "fontWeight": "500",
            "letterSpacing": 0,
            "lineHeight": 20,
          },
          Object {
            "marginBottom": 10,
          },
        ]
      }
    >
      Self-Reflection
    </Text>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#E6E8EB",
            "borderRadius": 100,
            "height": 8,
            "width": "90%",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgba(35, 175, 255, 1)",
            "borderRadius": 100,
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": "0%",
          }
        }
      />
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flexDirection": "column-reverse",
        "paddingBottom": 72,
      }
    }
    height={1334}
    style={
      Array [
        Object {
          "display": "flex",
          "height": 1199,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
          "width": "90%",
        },
      ]
    }
    width={750}
  >
    <View>
      <View
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "width": "100%",
            },
          ]
        }
      >
        <View
          marginBottom={0}
          marginTop={20}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "marginBottom": 0,
                "marginTop": 20,
                "maxWidth": 350,
                "width": 350,
              },
            ]
          }
          width={750}
        >
          <Text
            style={
              Array [
                Object {
                  "color": "rgb(36, 38, 43)",
                  "fontFamily": "Whitney-Book",
                  "fontSize": 18,
                  "fontWeight": "500",
                  "letterSpacing": 0,
                  "lineHeight": 18,
                },
              ]
            }
          >
            Rate your overall level of confidence based on the following sentence:
          </Text>
        </View>
        <View
          marginBottom={20}
          marginTop={20}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "marginBottom": 20,
                "marginTop": 20,
                "maxWidth": 350,
                "width": 350,
              },
            ]
          }
          width={750}
        >
          <View
            style={
              Object {
                "maxHeight": 0,
                "minHeight": 0,
                "opacity": 0,
                "width": "100%",
              }
            }
          >
            <View
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "justifyContent": "flex-start",
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "maxWidth": 350,
                      "width": "100%",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "rgb(36, 38, 43)",
                        "fontFamily": "Whitney-Book",
                        "fontSize": 20,
                        "fontWeight": "500",
                        "letterSpacing": 0,
                        "lineHeight": 20,
                        "textDecorationColor": "black",
                        "textDecorationLine": "none",
                        "textDecorationStyle": "solid",
                      },
                      Object {
                        "fontWeight": "700",
                        "marginBottom": 12,
                      },
                    ]
                  }
                >
                  I can manage to solve difficult problems if I try hard enough.
                </Text>
                <View>
                  <TouchableOpacity
                    accessibilityLabel="I am not at all confident"
                    accessibilityRole="button"
                    accessible={true}
                    alignLabel="left"
                    appearance="outline"
                    disabled={false}
                    intent="secondary"
                    onPress={[Function]}
                    size="medium"
                    style={
                      Array [
                        Object {
                          "alignItems": "flex-start",
                          "backgroundColor": "transparent",
                          "borderColor": "rgb(34, 110, 173)",
                          "borderRadius": 48,
                          "borderStyle": "solid",
                          "borderWidth": 2,
                          "display": "flex",
                          "justifyContent": "center",
                          "minHeight": 44,
                          "opacity": 1,
                          "paddingHorizontal": 22,
                        },
                        Object {
                          "marginBottom": 5,
                          "marginLeft": 0,
                          "marginRight": 0,
                          "marginTop": 5,
                        },
                      ]
                    }
                    width="none"
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "justifyContent": "center",
                          "minHeight": 20,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "position": "absolute",
                          }
                        }
                      >
                        <ActivityIndicator
                          animating={false}
                          color="rgb(34, 110, 173)"
                        />
                      </View>
                      <View
                        style={
                          Object {
                            "opacity": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "flexDirection": "row",
                              },
                              Object {
                                "alignItems": "center",
                              },
                            ]
                          }
                        >
                          <Text
                            appearance="outline"
                            disabled={false}
                            intent="secondary"
                            size="medium"
                            style={
                              Array [
                                Object {
                                  "color": "rgb(34, 110, 173)",
                                  "fontFamily": "Whitney-Semibold",
                                  "fontSize": 18,
                                  "fontWeight": "bold",
                                  "lineHeight": 18,
                                },
                              ]
                            }
                          >
                            I am not at all confident
                          </Text>
                        </View>
                      </View>
                    </View>
                  </TouchableOpacity>
                  <TouchableOpacity
                    accessibilityLabel="I am a little confident"
                    accessibilityRole="button"
                    accessible={true}
                    alignLabel="left"
                    appearance="outline"
                    disabled={false}
                    intent="secondary"
                    onPress={[Function]}
                    size="medium"
                    style={
                      Array [
                        Object {
                          "alignItems": "flex-start",
                          "backgroundColor": "transparent",
                          "borderColor": "rgb(34, 110, 173)",
                          "borderRadius": 48,
                          "borderStyle": "solid",
                          "borderWidth": 2,
                          "display": "flex",
                          "justifyContent": "center",
                          "minHeight": 44,
                          "opacity": 1,
                          "paddingHorizontal": 22,
                        },
                        Object {
                          "marginBottom": 5,
                          "marginLeft": 0,
                          "marginRight": 0,
                          "marginTop": 5,
                        },
                      ]
                    }
                    width="none"
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "justifyContent": "center",
                          "minHeight": 20,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "position": "absolute",
                          }
                        }
                      >
                        <ActivityIndicator
                          animating={false}
                          color="rgb(34, 110, 173)"
                        />
                      </View>
                      <View
                        style={
                          Object {
                            "opacity": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "flexDirection": "row",
                              },
                              Object {
                                "alignItems": "center",
                              },
                            ]
                          }
                        >
                          <Text
                            appearance="outline"
                            disabled={false}
                            intent="secondary"
                            size="medium"
                            style={
                              Array [
                                Object {
                                  "color": "rgb(34, 110, 173)",
                                  "fontFamily": "Whitney-Semibold",
                                  "fontSize": 18,
                                  "fontWeight": "bold",
                                  "lineHeight": 18,
                                },
                              ]
                            }
                          >
                            I am a little confident
                          </Text>
                        </View>
                      </View>
                    </View>
                  </TouchableOpacity>
                  <TouchableOpacity
                    accessibilityLabel="I am somewhat confident"
                    accessibilityRole="button"
                    accessible={true}
                    alignLabel="left"
                    appearance="outline"
                    disabled={false}
                    intent="secondary"
                    onPress={[Function]}
                    size="medium"
                    style={
                      Array [
                        Object {
                          "alignItems": "flex-start",
                          "backgroundColor": "transparent",
                          "borderColor": "rgb(34, 110, 173)",
                          "borderRadius": 48,
                          "borderStyle": "solid",
                          "borderWidth": 2,
                          "display": "flex",
                          "justifyContent": "center",
                          "minHeight": 44,
                          "opacity": 1,
                          "paddingHorizontal": 22,
                        },
                        Object {
                          "marginBottom": 5,
                          "marginLeft": 0,
                          "marginRight": 0,
                          "marginTop": 5,
                        },
                      ]
                    }
                    width="none"
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "justifyContent": "center",
                          "minHeight": 20,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "position": "absolute",
                          }
                        }
                      >
                        <ActivityIndicator
                          animating={false}
                          color="rgb(34, 110, 173)"
                        />
                      </View>
                      <View
                        style={
                          Object {
                            "opacity": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "flexDirection": "row",
                              },
                              Object {
                                "alignItems": "center",
                              },
                            ]
                          }
                        >
                          <Text
                            appearance="outline"
                            disabled={false}
                            intent="secondary"
                            size="medium"
                            style={
                              Array [
                                Object {
                                  "color": "rgb(34, 110, 173)",
                                  "fontFamily": "Whitney-Semibold",
                                  "fontSize": 18,
                                  "fontWeight": "bold",
                                  "lineHeight": 18,
                                },
                              ]
                            }
                          >
                            I am somewhat confident
                          </Text>
                        </View>
                      </View>
                    </View>
                  </TouchableOpacity>
                  <TouchableOpacity
                    accessibilityLabel="I am very confident"
                    accessibilityRole="button"
                    accessible={true}
                    alignLabel="left"
                    appearance="outline"
                    disabled={false}
                    intent="secondary"
                    onPress={[Function]}
                    size="medium"
                    style={
                      Array [
                        Object {
                          "alignItems": "flex-start",
                          "backgroundColor": "transparent",
                          "borderColor": "rgb(34, 110, 173)",
                          "borderRadius": 48,
                          "borderStyle": "solid",
                          "borderWidth": 2,
                          "display": "flex",
                          "justifyContent": "center",
                          "minHeight": 44,
                          "opacity": 1,
                          "paddingHorizontal": 22,
                        },
                        Object {
                          "marginBottom": 5,
                          "marginLeft": 0,
                          "marginRight": 0,
                          "marginTop": 5,
                        },
                      ]
                    }
                    width="none"
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "justifyContent": "center",
                          "minHeight": 20,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "position": "absolute",
                          }
                        }
                      >
                        <ActivityIndicator
                          animating={false}
                          color="rgb(34, 110, 173)"
                        />
                      </View>
                      <View
                        style={
                          Object {
                            "opacity": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "flexDirection": "row",
                              },
                              Object {
                                "alignItems": "center",
                              },
                            ]
                          }
                        >
                          <Text
                            appearance="outline"
                            disabled={false}
                            intent="secondary"
                            size="medium"
                            style={
                              Array [
                                Object {
                                  "color": "rgb(34, 110, 173)",
                                  "fontFamily": "Whitney-Semibold",
                                  "fontSize": 18,
                                  "fontWeight": "bold",
                                  "lineHeight": 18,
                                },
                              ]
                            }
                          >
                            I am very confident
                          </Text>
                        </View>
                      </View>
                    </View>
                  </TouchableOpacity>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
